staging_vespalib/src/vespa/vespalib/util/singleexecutor.cpp

```diff
@@ -60,7 +86,9 @@ SingleExecutor &
 SingleExecutor::sync() {
     uint64_t wp = _wp.load(std::memory_order_relaxed);
     while (wp > _rp.load(std::memory_order_acquire)) {
-        std::this_thread::sleep_for(1ms);
+        wakeupConsumer();
```

Comment:
note that this signal may be lost since no state is being passed with the signal (the other thread might call wait after the signal has been discarded since no-one was waiting).

Reply:
Yes, correct. The purpose is to wake up one that is sleeping waiting for low watermark to be reached.